app/src/main/java/com/github/se/eduverse/ui/archive/ArchiveScreen.kt

```diff
@@ -0,0 +1,77 @@
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.material3.CenterAlignedTopAppBar
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import com.github.se.eduverse.model.Folder
+import com.github.se.eduverse.ui.navigation.BottomNavigationMenu
+import com.github.se.eduverse.ui.navigation.LIST_TOP_LEVEL_DESTINATION
+import com.github.se.eduverse.ui.navigation.NavigationActions
+import com.github.se.eduverse.ui.navigation.Route
+import com.github.se.eduverse.viewmodel.FolderViewModel
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun ArchiveScreen(navigationActions: NavigationActions, folderViewModel: FolderViewModel) {
+  folderViewModel.getArchivedUserFolders()
+  val folders: List<Folder> by folderViewModel.folders.collectAsState()
+
+  Scaffold(
+      topBar = {
+        CenterAlignedTopAppBar(
+            modifier = Modifier.testTag("topAppBar"),
+            title = { Text(text = "Archived Courses", modifier = Modifier.testTag("topBarText")) },
+            navigationIcon = {
+              IconButton(
+                  onClick = { navigationActions.goBack() }, modifier = Modifier.testTag("goBack")) {
+                    Icon(
+                        imageVector = Icons.AutoMirrored.Filled.ArrowBack,
+                        contentDescription = "Back Arrow")
+                  }
+            },
+            colors =
+                TopAppBarDefaults.centerAlignedTopAppBarColors(containerColor = Color.Transparent))
+      },
+      bottomBar = {
+        BottomNavigationMenu({ navigationActions.navigateTo(it) }, LIST_TOP_LEVEL_DESTINATION, "")
+      }) { padding ->
+        Column(modifier = Modifier.padding(padding).fillMaxSize()) {
+          folders.forEach {
+            Card(
+                modifier =
+                    Modifier.padding(8.dp)
+                        .fillMaxWidth()
+                        .clickable {
+                          folderViewModel.unarchiveFolder(it)
+                          navigationActions.navigateTo(Route.LIST_FOLDERS)
+                        }
+                        .testTag("folderCard${it.id}"),
+                elevation = 4.dp) {
+                  Text(
+                      text = it.name,
+                      modifier = Modifier.padding(16.dp),
+                      style = androidx.compose.material.MaterialTheme.typography.h6)
+                }
+          }
+        }
+      }
```

Comment:
The logic that unarchives a folder and navigates is built directly into the Card composable. It's not mandatory/crucial (we've all done it) but it would be interesting to move this logic into an external function or into the ViewModel to separate the business logic from the UI.